src/components/Navbar.jsx

```diff
@@ -3,41 +3,79 @@ import { Menu, Header, Icon, Responsive, Sidebar } from 'semantic-ui-react'
 import '../css/navbar.css'
 import { NavLink } from 'react-router-dom'
 import { connect } from 'react-redux'
+import Login from './Login'
 import Logout from './Logout'
 
 class Navbar extends Component {
-  state = { visibleSidebar: false }
+  state = { visibleSidebar: false, modalOpen: false }
+
+  handleModalOpen = () => {
+    this.setState(prevState => {
+      return {
+        modalOpen: !prevState.modalOpen
+      }
+    })
+  }
 
   handleShowClick = () => this.setState({ visibleSidebar: true })
   handleSidebarHide = () => this.setState({ visibleSidebar: false })
 
   render() {
     const notMobile = { minWidth: Responsive.onlyMobile.maxWidth + 1 }
-    let logOut, logIn
+    let logOut, logIn, welcomeMessage, createRecipe
 
     if (this.props.currentUser.isSignedIn) {
+      welcomeMessage = (
+        <Menu.Item style={{ backgroundColor: 'white' }}>
+          <Header
+            position='right'
+            style={{ color: 'green' }}
+            id="welcome-message"
+          >
+            Hello {this.props.currentUser.attributes.name}
```

Comment:
Do we have a name attribute on a User object?

Reply:
Yes we do! :D

Reply:
Really? 🤪